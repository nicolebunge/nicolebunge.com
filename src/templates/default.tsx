import { graphql, PageProps } from 'gatsby';
import React from 'react';
import SEO from '../components/SEO';
import Section from '../components/Section';
import Grid from '../components/Grid';

interface DataType {
  markdownRemark: {
    frontmatter: {
      title: string;
    };
    html: string;
  };
}

export type DefaultTemplateProps = PageProps<DataType>;

function DefaultTemplate(props: DefaultTemplateProps): JSX.Element {
  const {
    data: {
      markdownRemark: { frontmatter, html },
    },
  } = props;

  return (
    <>
      <SEO title={frontmatter.title} />
      <Section>
        <Grid>
          {/* eslint-disable-next-line react/no-danger */}
          <div dangerouslySetInnerHTML={{ __html: html }} />
        </Grid>
      </Section>
    </>
  );
}

export const pageQuery = graphql`
  query ($slug: String!) {
    markdownRemark(frontmatter: { path: { eq: $slug } }) {
      html
      frontmatter {
        title
      }
    }
  }
`;

export default DefaultTemplate;
