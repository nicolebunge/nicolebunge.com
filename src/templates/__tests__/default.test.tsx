import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import React from 'react';
import * as stories from '../default.stories';

const { Default } = composeStories(stories);

jest.mock('../../hooks/site');

describe('DefaultTemplate component', () => {
  it('should render correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
