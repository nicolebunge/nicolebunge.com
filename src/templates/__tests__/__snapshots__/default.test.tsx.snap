// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultTemplate component should render correctly 1`] = `
<div
  class="layout"
>
  <header
    class="header"
    role="banner"
  >
    <a
      href="/"
    >
      <img
        alt="Site Title"
        height="28"
        loading="lazy"
        src="https://res.cloudinary.com/nicolebunge/image/upload/c_fill,f_auto,h_28,q_auto,w_160/v1627923793/nicolebunge.com/nicole-bunge.svg"
        width="160"
      />
    </a>
    <nav
      class="mainNavigation"
    >
      <button
        class="mainNavigation__button"
        type="button"
      >
        <svg
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24 6H0V2h24v4zm0 4H0v4h24v-4zm0 8H0v4h24v-4z"
          />
        </svg>
      </button>
      <ul
        class="mainNavigation__list mainNavigation__listInline"
      >
        <li
          class="mainNavigation__item"
        >
          <a
            class="mainNavigation__link"
            href="/ueber-mich/"
          >
            Über mich
          </a>
        </li>
        <li
          class="mainNavigation__item"
        >
          <a
            class="mainNavigation__link"
            href="/schauspiel/"
          >
            Schauspiel
          </a>
        </li>
        <li
          class="mainNavigation__item"
        >
          <a
            class="mainNavigation__link"
            href="/theaterpaedagogik/"
          >
            Theaterpädagogik
          </a>
        </li>
        <li
          class="mainNavigation__item"
        >
          <a
            class="mainNavigation__link"
            href="/galerie/"
          >
            Galerie
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="main"
    role="main"
  >
    <section
      class="section"
    >
      <div
        class="grid"
      >
        <div>
          <p>
            Page content
          </p>
        </div>
      </div>
    </section>
  </main>
  <footer
    class="footer"
    role="contentinfo"
  >
    <div
      class="grid"
    >
      <nav
        class="navigation"
      >
        <ul
          class="navigation__list navigation__listInline"
        >
          <li
            class="navigation__item"
          >
            <a
              class="navigation__link"
              href="/impressum/"
            >
              Impressum
            </a>
          </li>
          <li
            class="navigation__item"
          >
            <a
              class="navigation__link"
              href="/datenschutz/"
            >
              Datenschutz
            </a>
          </li>
        </ul>
      </nav>
      <p>
        © 
        2000
         Nicole Bunge
      </p>
    </div>
  </footer>
  <div
    class="social"
  >
    <ul
      class="social__icons"
    >
      <li>
        <a
          class="social__icon"
          href="https://www.instagram.com/nici.bunge"
          itemprop="sameAs"
          rel="noopener noreferrer"
          target="_blank"
          title="Nicole Bunge auf Instagram"
        >
          <svg
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.829 6.302c-.738-.034-.96-.04-2.829-.04s-2.09.007-2.828.04c-1.899.087-2.783.986-2.87 2.87-.033.738-.041.959-.041 2.828s.008 2.09.041 2.829c.087 1.879.967 2.783 2.87 2.87.737.033.959.041 2.828.041 1.87 0 2.091-.007 2.829-.041 1.899-.086 2.782-.988 2.87-2.87.033-.738.04-.96.04-2.829s-.007-2.09-.04-2.828c-.088-1.883-.973-2.783-2.87-2.87zM12 15.595a3.595 3.595 0 1 1 0-7.19 3.595 3.595 0 0 1 0 7.19zm3.737-6.491a.84.84 0 1 1 0-1.68.84.84 0 0 1 0 1.68zM14.333 12a2.333 2.333 0 1 1-4.666 0 2.333 2.333 0 0 1 4.666 0zM12 0C5.373 0 0 5.373 0 12s5.373 12 12 12 12-5.373 12-12S18.627 0 12 0zm6.958 14.886c-.115 2.545-1.532 3.955-4.071 4.072-.747.034-.986.042-2.887.042s-2.139-.008-2.886-.042c-2.544-.117-3.955-1.529-4.072-4.072C5.008 14.14 5 13.901 5 12c0-1.901.008-2.139.042-2.886.117-2.544 1.529-3.955 4.072-4.071C9.861 5.008 10.099 5 12 5s2.14.008 2.887.043c2.545.117 3.957 1.532 4.071 4.071.034.747.042.985.042 2.886 0 1.901-.008 2.14-.042 2.886z"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          class="social__icon"
          href="https://www.linkedin.com/in/nicolebunge"
          itemprop="sameAs"
          rel="noopener noreferrer"
          target="_blank"
          title="Nicole Bunge auf LinkedIn"
        >
          <svg
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0C5.373 0 0 5.373 0 12s5.373 12 12 12 12-5.373 12-12S18.627 0 12 0zm-2 16H8v-6h2v6zM9 9.109c-.607 0-1.1-.496-1.1-1.109 0-.612.492-1.109 1.1-1.109s1.1.497 1.1 1.109c0 .613-.493 1.109-1.1 1.109zM17 16h-1.998v-2.861c0-1.881-2.002-1.722-2.002 0V16h-2v-6h2v1.093c.872-1.616 4-1.736 4 1.548V16z"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          class="social__icon"
          href="https://www.xing.com/profile/Nicole_Bunge3"
          itemprop="sameAs"
          rel="noopener noreferrer"
          target="_blank"
          title="Nicole Bunge auf XING"
        >
          <svg
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0C5.373 0 0 5.373 0 12s5.373 12 12 12 12-5.373 12-12S18.627 0 12 0zM6.089 14.996l2.4-4.233-1.614-2.767h2.881l1.601 2.759-2.401 4.241H6.089zM16.939 19h-2.848l-3.105-5.639L15.694 5h2.877l-4.707 8.361L16.939 19z"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
